src/Authentication/Wsse.php

```diff
@@ -23,24 +23,33 @@ final class Wsse implements Authentication
     private $password;
 
     /**
-     * @param string $username
-     * @param string $password
+     * @var bool
      */
-    public function __construct($username, $password)
+    private $useSha512;
+
+    /**
+     * @param string     $username
+     * @param string     $password
+     * @param bool|false $useSha512
+     */
+    public function __construct($username, $password, $useSha512 = false)
     {
         $this->username = $username;
         $this->password = $password;
+        $this->useSha512 = $useSha512;
     }
 
     /**
      * {@inheritdoc}
      */
     public function authenticate(RequestInterface $request)
     {
-        // TODO: generate better nonce?
         $nonce = substr(md5(uniqid(uniqid().'_', true)), 0, 16);
         $created = date('c');
         $digest = base64_encode(sha1(base64_decode($nonce).$created.$this->password, true));
```

Comment:
i think this should be an if/else to not unnecessarly do the sha1 encoding and then throw it away immediately